src/main/java/com/google/cloud/tools/managedcloudsdk/install/ExtractorProvider.java

```diff
@@ -28,8 +27,7 @@
    *
    * @param archive the archive to extract
    * @param destination the destination folder for extracted files
-   * @param messageListener a listener for extraction messages
    * @throws IOException if extractor fails
    */
-  void extract(Path archive, Path destination, MessageListener messageListener) throws IOException;
+  void extract(Path archive, Path destination) throws IOException;
```

Comment:
Since you're changing this signature, might it be better to add the ProgressListener here now, even if the implementors do nothing with it?

Reply:
This signature can change later without the implementers API changing.